### PR TITLE
Bug fix: exception in sender on device token failure

### PR DESCRIPTION
**Pull request by Guriido**
- Add a test case to check this problem is fixed
- bug fix

### CODE REVIEW

File: tests/test_sender.py [hunk 1]
```diff
@@ -160,3 +171,110 @@ def test_sender_e2e(monkeypatch: MonkeyPatch):
...
+    dated_image = ("", Image.new(mode="RGB", size=(200, 200)))
+    sender.start()
+    sender.enqueue(dated_image)
+    time.sleep(1)
+    sender.enqueue(dated_image)
+    time.sleep(0.5)
+    sender.stop()
+    sender.join(10)
+
+    sender
```
Comment (senz1024) [inline tests/test_sender.py:199]: Is this necessary?

Comment (Guriido) [reply tests/test_sender.py:199]: Do you imply that the wait is too long?
At first, I used a join without a timeout, but if the sender throws an exception (thus, the test should fail), the test hangs on the join...
So I set a timeout of 10 sec, but in practice, the wait only happens on a failing test.

Comment (senz1024) [reply tests/test_sender.py:199]: No, I meant what L199 does. :pray:

Comment (Guriido) [reply tests/test_sender.py:199]: it does absolutely nothing :innocent: 
fixing it right away

Comment (Guriido) [reply tests/test_sender.py:199]: [`ccd908a` (#18)](https://github.com/Idein/learning-pipeline-plugin/pull/18/commits/ccd908a0d7537c596455bc47e7ffbba63489ebaa)

Comment (senz1024) [reply tests/test_sender.py:199]: L199 still remains? (There were two same lines :innocent: )

Comment (Guriido) [reply tests/test_sender.py]: [`920427a` (#18)](https://github.com/Idein/learning-pipeline-plugin/pull/18/commits/920427ae994d304f93e3aab9588a8ee0c840a687) :flushed:

**Review by senz1024 — APPROVED** [review]
LGTM!

**Pull request merged**